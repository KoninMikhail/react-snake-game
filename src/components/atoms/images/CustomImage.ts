import styled from 'styled-components';

/**
 * @typedef ICustomImageProps
 *
 * @property {string} src - The image src
 * @property {string} width - The image width (css rules)
 * @property {string=} height - The image height (css rules)
 * @property {string} alt - The image alt
 */
interface ICustomImageProps {
  src: string;
  width: string;
  height?: string;
  alt?: string;
}

/**
 * CustomImage
 *
 * @description Use for image on the page
 *
 * @param {ICustomImageProps} - Component properties
 * @returns {JSX.Element}
 */
export const CustomImage = styled.img.attrs<ICustomImageProps>((props) => ({
  src: props.src,
  alt: props.alt || '',
}))`
  width: ${(props) => props.width};
  height: ${(props) => props.height || 'auto'};
`;
