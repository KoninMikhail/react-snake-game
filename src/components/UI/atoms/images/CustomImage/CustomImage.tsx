import { StyledCustomImage } from './StyledCustomImage';

/**
 * @typedef ICustomImageProps
 *
 * @property {string} src - The image src
 * @property {string} width - The image width (css rules)
 * @property {string=} height - The image height (css rules)
 * @property {string} alt - The image alt
 */
export interface ICustomImageProps {
  src: string;
  width: string;
  height?: string;
  alt?: string;
}

/**
 * CustomImage
 *
 * Use for image on the page
 *
 * @param {ICustomImageProps} props- Component properties
 * @returns {JSX.Element}
 */
export const CustomImage = (props: ICustomImageProps): JSX.Element => (
  <StyledCustomImage {...props} />
);
