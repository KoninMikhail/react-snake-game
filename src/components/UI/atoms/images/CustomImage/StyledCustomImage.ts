import styled from 'styled-components';
import { ICustomImageProps } from './CustomImage';

export const StyledCustomImage = styled.img.attrs<ICustomImageProps>((props) => ({
  src: props.src,
  alt: props.alt || '',
}))`
  width: ${(props) => props.width};
  height: ${(props) => props.height || 'auto'};
`;
