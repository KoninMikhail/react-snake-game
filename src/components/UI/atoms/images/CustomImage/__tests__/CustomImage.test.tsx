import { render, screen } from '@testing-library/react';
import { CustomImage } from '../CustomImage';

describe('Custom Image', () => {
  const imageUrl = '#imageurl';
  const imageTestId = 'custom-image';

  it('should create a image with only require attrs', () => {
    render(<CustomImage src={imageUrl} width={'62vw'} data-testid={imageTestId} />);
    const customImageElement = screen.getByTestId(imageTestId);

    expect(customImageElement).toHaveAttribute('src', '#imageurl');
    expect(customImageElement).toHaveAttribute('width', '62vw');
  });

  it('should render the image with all attrs', () => {
    render(
      <CustomImage
        src={imageUrl}
        width={'76vw'}
        height={'22vh'}
        alt={'alt'}
        data-testid={imageTestId}
      />
    );
    const customImageElement = screen.getByTestId(imageTestId);

    expect(customImageElement).toHaveAttribute('src', '#imageurl');
    expect(customImageElement).toHaveAttribute('width', '76vw');
    expect(customImageElement).toHaveAttribute('height', '22vh');
    expect(customImageElement).toHaveAttribute('alt', 'alt');
  });

  it('render correctly', () => {
    render(
      <CustomImage
        src={imageUrl}
        width={'76vw'}
        height={'22vh'}
        alt={'alt'}
        data-testid={imageTestId}
      />
    );

    expect(screen.getByTestId(imageTestId)).toMatchSnapshot();
  });
});
