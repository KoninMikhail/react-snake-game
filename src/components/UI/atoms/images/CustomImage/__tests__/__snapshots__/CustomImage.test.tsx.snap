// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Image render correctly 1`] = `
.c0 {
  width: 76vw;
  height: 22vh;
}

<img
  alt="alt"
  class="c0"
  data-testid="custom-image"
  height="22vh"
  src="#imageurl"
  width="76vw"
/>
`;
